sympy/matrices/common.py

```diff
@@ -2018,10 +2018,12 @@ def __mul__(self, other):
         if getattr(other, 'is_MatrixLike', False):
             return MatrixArithmetic._eval_matrix_mul(self, other)
 
-        try:
-            return self._eval_scalar_mul(other)
-        except TypeError:
-            pass
+        # if 'other' is not iterable then scalar multiplication.
+        if not isinstance(other, collections.Iterable):
+            try:
+                return self._eval_scalar_mul(other)
+            except TypeError:
+                pass
 
         raise TypeError('Cannot multiply %s and %s' % (type(self), type(other)))
```

Comment:
I think it's better to return `NotImplemented`, as Python automatically constructs and raises `TypeError` if `NotImplemented` is returned by overridden Python operations.

Reply:
Yes. `NotImplemented` is better. But I wonder in which case it will not be implemented ?

Reply:
As you said, in cases of `Matrix*Iterable`.

Reply:
but `iterable` will be omitted after that `if` statement.

Reply:
oh sorry I wanted to ask about line 2025 `except`. In which case code will raise `TypeError`.

Reply:
Multiplication that makes no sense like `None*Matrix` would raise it.

Reply:
Ok Thanks.